.github/workflows/build-irods-http-api-ubuntu.yml

```diff
@@ -0,0 +1,62 @@
+name: iRODS HTTP API Builder
+on: [pull_request]
+jobs:
+  build:
+    name: Build - ${{ matrix.os }}, iRODS ${{ matrix.irods-version }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-20.04, ubuntu-22.04, ubuntu-24.04 ]
```

Comment:
Might have to remove Ubuntu-20.04, or it grows its own file if there's a fix for this:

```console
$ sudo apt-get update -qq
$ sudo apt-get install -qq \
>    apt-transport-https \
>    git \
>    gnupg \
>    lsb-release \
>    wget \
>    g++-11 \
>    gcc-11 \
>    libcurl4-gnutls-dev \
>    libssl-dev \
>    libssl3 \
>    ninja-build

E: Unable to locate package gcc-11
E: Package 'libssl3' has no installation candidate
```

Reply:
Could try using `uname` or `/etc/os-release` to sniff the OS version and install different packages based on the results. That's if you want to keep it to one file.

Creating a new file is fine too, assuming GitHub won't complain about duplicate workflow names.

Reply:
In Discuss:

We will remove Ubuntu 20.04.

Reply:
ub20 is not EOL until april 2025?

Reply:
Maybe relevant: https://github.com/irods/irods_client_http_api/pull/370#discussion_r1868513336

Reply:
Removed ub20, not resolving, so there can be discussion.

Reply:
Discuss:

Will add ub20, @korydraughn will help out with figuring out the right packages.

Reply:
Use the following:
- `libssl1.1`
- `g++-10`
- `gcc-10`

Reply:
You'll likely get a linker error for `dlclose`. Not sure why I don't get that error when building directly on the host OS.

Will open an issue for that soon.

Reply:
Should I add the build for 20 here, and then we wait on the issue? Or have 20 separate, like the other images?

Reply:
Let's add ub20 to this since we're handling the others.
- Add a new commit for the linker error
- Then, add a github action for ub20

Reply:
Rebased onto branch that has fix.

Also added ub20.

Done.